### ISSUE TITLE
Set-ServiceCredential not called?

### ISSUE THREAD

**Issue by kottimon** [OctopusDeploy/OctopusDSC#84]
Hi!

Sorry if this should not be posted as an issue since I've solved it for myself, new to GIT...

I'm trying to install a new tentacle with service credentials (encrypted with certificate), but I'm running in to a problem.
The function Set-ServiceCredential is never called in cTentacleAgent.psm1.

I've added this part to Set-TargetResource to make it work:

```
    if ($TentacleServiceCredential)
    {
        Write-Verbose "Settings servicecredentials."
        Set-ServiceCredential -Name $name -credential $TentacleServiceCredential
        Write-Verbose "Restarting $serviceName with new credentials."
        Restart-Service -Name $serviceName
    }
    else {
        Write-Verbose "Skipping service credentials as no credentials givven."
    }
```

And this seems to have done the trick! :)

**Comment by matt-richardson** [comment]
Good catch, @kottimon. Seems our tests weren't catching what we thought they were.

Both the tests and the code have been fixed now - its available in `3.0.70`.

**Comment by kottimon** [comment]
That's great, thanks!

**Comment by kottimon** [comment]
Hello again @matt-richardson  :)

It seems that there is a typo in the new code:
```
    if ($TentacleServiceCredential) {
        Write-Log "Adding Service identity to installation command"

        $serviceArgs += @(
            '--username', $TentacleServiceCredential.UserName
            '--password', $TentacleServiceCredential.GetNetworkCredential().Password
        )
    } 
```
The term 'Write-Log' is not recognized as the name of a cmdlet, function...
I'm guessing it should be write-verbose there as in the other parts, and write-log was something from your debugging.

**Comment by matt-richardson** [comment]
Ahh, sorry about that, @kottimon. Bit of a copy pasta error there...
Just pushed a fix - once all the tests pass, it should publish in about half an hour.

**Comment by kottimon** [comment]
No worries :)

**Comment by matt-richardson** [comment]
This is fixed in `3.0.74`.
